src/mongocrypt-marking.c

```diff
@@ -942,6 +1076,7 @@ _mongocrypt_fle2_placeholder_to_find_ciphertext (
    mc_FLE2FindEqualityPayload_t payload;
    bool res = false;
 
+   BSON_ASSERT (kb->crypt->opts.use_fle2_v2 == false);
```

Comment:
```suggestion
```

I expect FLE1 payloads can still be produced when the feature flag is enabled.

Reply:
Yes, `use_fle2_v2` is more like `use_fle2_v2_when_processing_fle2_placeholders`, but that's a bit wordy. ;)